src/StoreKeeper/WooCommerce/B2C/Backoffice/MetaBoxes/ProductSyncMetaBox.php

```diff
@@ -0,0 +1,145 @@
+        }
+    }
+
+    final public function renderSyncBox($post): void
+    {
+        $product = wc_get_product($post->ID);
+        $syncUrl = esc_attr($this->getNonceSyncActionLink($post));
+        $storekeeperId = $this->getPostMeta($product->get_id(), 'storekeeper_id', false);
+
+        $idLabel = esc_html__('Backoffice ID', I18N::DOMAIN);
+        $idValue = esc_html($storekeeperId ?: '-');
+
+        $dateLabel = esc_html__('Last sync', I18N::DOMAIN);
+        $dateValue = esc_html($this->getPostMeta($product->get_id(), 'storekeeper_sync_date', '-'));
+
+        $backoffice = '';
+        if (0 !== (int) $storekeeperId && StoreKeeperOptions::isConnected()) {
+            $backofficeLabel = esc_html__('Open in backoffice', I18N::DOMAIN);
+            $backofficeLink = esc_attr(StoreKeeperOptions::getBackofficeUrl()."#products/details/$storekeeperId");
+            $backoffice = <<<HTML
+                        <a href="$backofficeLink" class="product_backoffice_link" target="_blank">$backofficeLabel</a>
+                        HTML;
+        }
+
+        $submitLabel = esc_html__('Force sync', I18N::DOMAIN);
+        echo <<<HTML
+            <ul class="product_actions submitbox">
+                <li class="wide">
+                    <div>   
+                        <strong>$idLabel:</strong>
+                        <div>$idValue</div>
+                    </div>
+                    <div>
+                        <strong>$dateLabel:</strong>
+                        <div>$dateValue</div>
+                    </div>
+                </li>
+                <li class="wide">        
+                    <a href="$syncUrl" class="button-primary product_sync_submission">$submitLabel</a>
+                    $backoffice
+                </li>
+            </ul>
+            HTML;
+
+        wp_enqueue_style('order-meta-box', plugin_dir_url(__FILE__).'../static/meta-boxes.css');
+        $this->showPossibleError();
+        $this->showPossibleSuccess();
+    }
+
+    /**
+     * Function to sync product from storekeeper backoffice to wordpress.
+     *
+     * @throws Exception
+     */
+    final public function doSync(int $postId): void
+    {
+        if ($this->isNonceValid($postId)) {
+            if ($product = wc_get_product($postId)) {
+                $storekeeperId = $this->getPostMeta($product->get_id(), 'storekeeper_id', false);
+                $tasks = $this->getTasks($storekeeperId);
+
+                if ($product->is_type('variable')) {
+                    $variationPostIds = $product->get_children();
+                    // Get all tasks related to variations and add to tasks
+                    foreach ($variationPostIds as $variationPostId) {
+                        $variationStorekeeperId = $this->getPostMeta($variationPostId, 'storekeeper_id', false);
+                        $taskMerger = $tasks;
+                        $tasks = array_merge($taskMerger, $this->getTasks($variationStorekeeperId));
```

Comment:
What is the reason to introduce extra varaible?
```suggestion
                        $tasks = array_merge($tasks, $this->getTasks($variationStorekeeperId));
```

Reply:
Because PHPStorm inspector is throwing a `is used in a loop and is a resources greedy construction.` if the `$tasks` variable  itself is used, but it can be bypassed if needed be.

Reply:
But now it even uses more resources :P cos you first make a copy of it

Reply:
maybe so something like 
```
foreach($this->getTasks($variationStorekeeperId) as  $task){
 $tasks[] = $task
}